R/MixedModelsCommon.R

```diff
@@ -563,761 +794,1135 @@
-    trendsSummary$addColumnInfo(name = "upperCI",  title = gettext("Upper"), type = "number",
-                             overtitle = gettextf("%s%% CI", 100 * options$trendsCIwidth))
-    if(options$trendsCompare){
-      trendsSummary$addColumnInfo(name = "stat",   title = ifelse(colnames(emm_test)[ncol(emm_test)-1] == "t.ratio", "t", "Z"), type = "number")
-      trendsSummary$addColumnInfo(name = "pval",   title = "p\u207A", type = "pvalue")
-      trendsSummary$addFootnote(.mmMessageTestNull(options$trendsCompareTo), symbol = "\u207A")
-      
-      if(options$pvalVS){
-        trendsSummary$addColumnInfo(name = "pvalVS", title = "VS-MPR\u002A", type = "number")
-        trendsSummary$addFootnote(.mmMessageVovkSellke, symbol = "\u002A")
-      }  
-    }
-  }else if(type %in% c("BLMM", "BGLMM")){
-    trendsSummary$addColumnInfo(name = "lowerCI",  title = gettext("Lower"), type = "number",
-                             overtitle = gettextf("%s%% HPD", 100 * options$trendsCIwidth))
-    trendsSummary$addColumnInfo(name = "upperCI",  title = gettext("Upper"), type = "number",
-                             overtitle = gettextf("%s%% HPD", 100 * options$trendsCIwidth))
-  }
-  
-  
-  for(i in 1:nrow(emm_table)){
+    if (type == "GLMM") {
+      EMMsummary$addFootnote(
+        ifelse(
+          options$marginalMeansResponse,
+          .mmMessageResponse,
+          .mmMessageNotResponse
+        )
+      )
+    }
+    
+    jaspResults[["EMMsummary"]] <- EMMsummary
+    jaspResults[["EMMresults"]] <- EMMresults
     
-    temp_row <- list()
     
-    if(options$trendsContrast){
-      temp_row$number <- i
+    object <- list(emm         = emm,
+                   emm_table   = emm_table)
+    EMMresults$object <- object
+    
+    return()
+  }
+.mmTrends        <-
+  function(jaspResults, dataset, options, type = "LMM") {
+    if (!is.null(jaspResults[["contrasts_Trends"]]))
+      return()
+    
+    model <- jaspResults[["mmModel"]]$object$model
+    
+    # deal with continuous predictors
+    at <- NULL
+    for (var in unlist(options$trendsVariables)) {
+      if (typeof(dataset[, .v(var)]) == "double") {
+        at[[.v(var)]] <-
+          c(
+            mean(dataset[, .v(var)], na.rm = TRUE) + c(-1, 0, 1) * options$trendsSD *
+              sd(dataset[, .v(var)], na.rm = TRUE)
+          )
+      }
     }
     
-    for(vi in 1:length(trends_var_names)){
-      if(typeof(dataset[,.v(trends_var_names[vi])]) == "double"){
-        temp_row[trends_var_names[vi]] <- emm_table[i,vi]
-      }else{
-        temp_row[trends_var_names[vi]] <- as.character(emm_table[i,vi])        
+    # compute the results
+    if (type %in% c("LMM")) {
+      emmeans::emm_options(pbkrtest.limit = if (options$trendsOverride)
+        Inf,
+        mmrTest.limit  = if (options$trendsOverride)
+          Inf)
+    }
+    
+    # TODO: deal with the emtrends scoping problems
```

Comment:
why do you need to put this stuff in the parent environment?

Reply:
I'm not sure what do you mean by that. I found this way of passing this two arguments in emmeans documentation